cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -555,13 +554,15 @@ protected StreamSizeAggregator(Id.Stream streamId, long baseCount, int streamThr
       this.streamInitSize = baseCount;
       this.streamThresholdMB = new AtomicInteger(streamThresholdMB);
       this.cancellable = cancellable;
-      this.isInit = true;
       this.streamId = streamId;
       this.streamFeed = new Id.NotificationFeed.Builder()
         .setNamespaceId(streamId.getNamespaceId())
         .setCategory(Constants.Notification.Stream.STREAM_FEED_CATEGORY)
         .setName(String.format("%sSize", streamId.getName()))
         .build();
+
+      // Publish an initialization notification
+      publishNotification(baseCount);
```

Comment:
it's usually not a good idea to do any work in a constructor, especially if you're connecting to other systems. There is a hidden requirement for the other system to be up and running when you create the object. I think it's better just to call publish explicitly after you create it, especially if this is a private class.  What happens if the publish fails? Do we still want to do the init check in case it the initial notification failed or are we ok with dropping notifications?